### PR TITLE
Style cleanup on tf2_py.cpp

### PR DESCRIPTION
**Pull request by ahcorde**
https://github.com/ros2/geometry2/issues/168

### CODE REVIEW

**Review by clalancette — APPROVED** [review]
One thing to note: in the past we had been avoiding these kinds of cleanups in https://github.com/ros2/geometry2 in the hope that we could share code with https://github.com/ros/geometry2 .  I think these two forks have diverged enough that that is no longer a viable option, so I'm approving.  But before merging please get confirmation from @tfoote that this is OK.

File: tf2_py/test/test_buffer_core.py [hunk 1]
```diff
@@ -1,34 +1,17 @@
-# Copyright 2019 Open Source Robotics Foundation, Inc.
-# All rights reserved.
+# Copyright 2016 Open Source Robotics Foundation, Inc.
 #
-# Redistribution and use in source and binary forms, with or without
-# modification, are permitted provided that the following conditions
-# are met:
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (tfoote) [inline tf2_py/test/test_buffer_core.py]: This is changing the license and the year. That doesn't look right.

File: tf2_py/test/__init__.py [hunk 2]
```diff
@@ -1,32 +1,16 @@
 # Copyright 2019 Open Source Robotics Foundation, Inc.
-# All rights reserved.
 #
-# Redistribution and use in source and binary forms, with or without
-# modification, are permitted provided that the following conditions
-# are met:
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (tfoote) [inline tf2_py/test/__init__.py]: This is changing the license not linting it.

File: tf2_py/src/tf2_py.cpp [hunk 3]
```diff
@@ -1,3 +1,17 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
```
Comment (tfoote) [inline tf2_py/src/tf2_py.cpp]: This should have the BSD license to match the origin's license.

File: tf2_py/src/python_compat.h [hunk 4]
```diff
@@ -1,5 +1,18 @@
 #ifndef TF2_PY_PYTHON_COMPAT_H
 #define TF2_PY_PYTHON_COMPAT_H
+// Copyright 2020 Open Source Robotics Foundation, Inc.
```
Comment (tfoote) [inline tf2_py/src/python_compat.h]: This should have the BSD header as it's from the legacy code base.

File: tf2_py/tf2_py/__init__.py [hunk 5]
```diff
@@ -1,36 +1,35 @@
...
-#*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
-#*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-#*  POSSIBILITY OF SUCH DAMAGE.
-#* 
-#* Author: Eitan Marder-Eppstein
-#***********************************************************
+# Copyright 2009 Willow Garage, Inc.
+# All rights reserved.
+#
+# Software License Agreement (BSD License 2.0)
```
Comment (tfoote) [inline tf2_py/tf2_py/__init__.py]: What's the BSD License 2.0? This is the 3-Clause BSD if you want a proper name.

**Review by tfoote — CHANGES_REQUESTED** [review]
I think we've diverged enough to let this cleanup happen. 

I had a few notes about the license headers though.

**Comment by ahcorde** [comment]
@tfoote  I have reviewed again the licenses

I have seen that the licenses are The 3-Clause BSD License. But the only bsd license accepted is BSD 2.0 according with the code ament_copyright https://github.com/ament/ament_lint/blob/master/ament_copyright/ament_copyright/licenses.py#L41

**Comment by tfoote** [comment]
I don't know where @clalancette got that copy of the license for https://github.com/ament/ament_lint/commit/767257ff141d7957f3c8c7b92106aced95794537 

There's no such things at "Version 2" listed at https://spdx.org/licenses/BSD-3-Clause.html or https://opensource.org/licenses/BSD-3-Clause or as is in the code. That looks like an accidental copy and paste misstep. 

In general though the linter needs to be able to accept any valid license. We can do minor whitespace updates if necessary but otherwise we cannot change the verbiage in the license on anything except what we've originally written. We talked about this a while ago and knew that the linters were too strict for applying to imported code. But we chose just not to turn them on instead of making them much more generic for expediency. If we want to turn them on for legacy code they will need to be updated to be more flexible.

**Comment by clalancette** [comment]
> I don't know where @clalancette got that copy of the license for [ament/ament_lint@767257f](https://github.com/ament/ament_lint/commit/767257ff141d7957f3c8c7b92106aced95794537)
> 
> There's no such things at "Version 2" listed at https://spdx.org/licenses/BSD-3-Clause.html or https://opensource.org/licenses/BSD-3-Clause or as is in the code. That looks like an accidental copy and paste misstep.

Yeah, I'm honestly not sure where that came from now that I look at it.  https://github.com/ament/ament_lint/issues/109 addresses some of the same issue.

> In general though the linter needs to be able to accept any valid license. We can do minor whitespace updates if necessary but otherwise we cannot change the verbiage in the license on anything except what we've originally written. We talked about this a while ago and knew that the linters were too strict for applying to imported code.

https://github.com/ament/ament_lint/issues/75 (somewhat).

> But we chose just not to turn them on instead of making them much more generic for expediency. If we want to turn them on for legacy code they will need to be updated to be more flexible.

For this PR, I'll suggest that we go back to the old copyright for now, and just disable the copyright check.  Once we figure out the ament_lint situation, we can come back and turn it on.

**Comment by ahcorde** [comment]
@clalancette, 

Excluded ament_copyright

should I include the license in the C++ code? beacuse cpplint is also failing because of the licence

**Comment by clalancette** [comment]
> should I include the license in the C++ code? beacuse cpplint is also failing because of the licence

I would say let's turn off cpplint for now as well.  Once we resolve the situation in ament_lint, we can revisit this and reenable those.

**Comment by ahcorde** [comment]
Excluded cpplint too, should I run the CI?

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9199/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4896/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7502/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9118/)
* Windows-container [Build Status](http://ci.ros2.org/job/ci_windows-container/5/)

**Review by clalancette — APPROVED** [review]

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9308/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5000/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7618/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9234/)
* Windows-container [Build Status](http://ci.ros2.org/job/ci_windows-container/111/)

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9371/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5054/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7673/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9280/)
* Windows-container [Build Status](http://ci.ros2.org/job/ci_windows-container/154/)

**Comment by dirk-thomas** [comment]
The change broke all jobs running on Ubuntu Focal: http://build.ros2.org/view/Fci/

```
geometry2/tf2_py/src/tf2_py.cpp:1059:1: error: cannot convert ‘std::nullptr_t’ to ‘Py_ssize_t’ {aka ‘long int’} in initialization
 1059 | };
      | ^
```

@ahcorde Please fix asap.

**Pull request merged**